aruco_tracking:
  process_freq: 50

  # flag
  compare_mode: false

  # threshold
  filter_buf_size: 20 # moving average filter buff size
  noise_dist_th_m: 3.0
  noise_cnt_th: 15 # (3/10)*process_frequency

  dft_buf_size: 256
  dft_integral_start_point: 100
  frequency_degree_buf_size: 20
  frequency_degree_threshold: 20
  difference_threshold_m: 5

  # aruco marker
    #---------------------
    # Estimating method
    # 0. Without filter
    # 1. Moving average filter
    # 2. Exponential moving average filter
    #---------------------
  estimating_method: 2
  dictionaryID: 0
  # marker_size_type_num: 6
  # marker_ids_type_0: [0]
  # marker_ids_type_1: [1]
  # marker_ids_type_2: [2]
  # marker_ids_type_3: [3]
  # marker_ids_type_4: [4]
  # marker_ids_type_5: [5]
  # marker_sizes_m: [0.72, 0.445, 0.275, 0.17, 0.105, 0.0648]
  marker_size_type_num: 3
  marker_ids_type_0: [0, 1, 2, 3, 4, 5, 6, 7]
  marker_ids_type_1: [8, 9, 10, 11, 12, 13, 14, 15]
  marker_ids_type_2: [16, 17, 18, 19]
  marker_sizes_m: [0.30, 0.09, 0.0416]
  big_marker_trans: 0.35
  medium_marker_trans: 0.105
  small_marker_trans: 0.0243
  plane_threshold: 0.1
  plane_iterations: 1000
  angle_deg_threshold: 28
  
  # frame id
  camera_frame_id: 'camera_link'

aruco_tf:
  process_freq: 50.0

aruco_visual:
  # marker_size_type_num: 6
  # marker_ids_type_0: [0]
  # marker_ids_type_1: [1]
  # marker_ids_type_2: [2]
  # marker_ids_type_3: [3]
  # marker_ids_type_4: [4]
  # marker_ids_type_5: [5]
  marker_size_type_num: 3
  marker_ids_type_0: [0, 1, 2, 3, 4, 5, 6, 7]
  marker_ids_type_1: [8, 9, 10, 11, 12, 13, 14, 15]
  marker_ids_type_2: [16, 17, 18, 19]